independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/mapping/RequestMapper.java

```diff
@@ -48,9 +49,12 @@ public void accept(String stem, ArrayList<RequestPath<T>> list) {
      * @return best RequestMatch, or null if the path has no match
      */
     public RequestMatch<T> map(String path) {
-        var result = mapFromPathMatcher(path, requestPaths.match(path), 0);
-        if (result != null) {
-            return result;
+        List<PathMatcher.PathMatch<ArrayList<RequestPath<T>>>> matches = requestPaths.match(path);
+        for (PathMatcher.PathMatch<ArrayList<RequestPath<T>>> match : matches) {
```

Comment:
@geoand I know that in some hot paths, you went back to simple indexed for loops. I wonder if this is such a hot path (given my understanding is that it's going to be called for each request).

Reply:
Yeah, exactly. Memory allocation occurs when using the iterator. For most intents and purposes, the constructs are the same (and for arrays, they are **exactly** the same), but when on the hot path, we do try to save as many allocations as possible.

Reply:
Got it, thanks for the insight :).